webapp/lib/view.dart

```diff
@@ -730,28 +730,43 @@ class ConversationListPanelView {
       _conversationList.removeItem(summary);
       return true;
     });
+    List<ConversationSummary> conversationsToAdd = [];
     for (var conversation in conversations) {
-      addOrUpdateConversation(conversation);
+      ConversationSummary summary = _phoneToConversations[conversation.docId];
+      if (summary != null) {
+        updateConversationSummary(summary, conversation);
+        continue;
+      }
+      summary = new ConversationSummary(
+          conversation.docId,
+          conversation.messages.first.text,
+          conversation.unread);
+      conversationsToAdd.add(summary);
+    }
+    _conversationList.appendItems(conversationsToAdd);
+    for (var conversation in conversationsToAdd) {
+      _phoneToConversations[conversation.deidentifiedPhoneNumber] = conversation;
     }
+    _conversationPanelTitle.text = '${_phoneToConversations.length} conversations';
   }
 
   void addOrUpdateConversation(Conversation conversation) {
```

Comment:
Is this method used anymore?
Can it be removed now that you have improved the `updateConversationList` above?

Reply:
It's not used anymore, and I considered removing it, but I think I'd like to keep it around for a bit longer in case we end up needing it. I'll set a timer on it for a couple of months from now, and if we haven't used it by then, we can remove it.